### PR TITLE
Clean up pt. 1

### PR DESCRIPTION
**Pull request by hanpeter**
## What does this PR do?

1. This PR re-organizes the requirements of this library. It breaks out tests requirements on its own separate section.
2. This PR updates and cleans out many outdated info in `setup.py`.
3. This PR removes the documentation that is no longer being maintained.
4. This PR removes `requirements.pip` and changes the repo to depend fully on `setup.py`.

## Why is this change being made?

1. This repo is in need of some TLC.
2. This PR is a suggestion to remove `requirements.pip`.

## How was this tested? How can the reviewer verify your testing?

The change was tested on the local development environment using following two commands:
* `$ pip install -e .`
* `$ nosetests`

## Completion checklist

- [x] The version of the package has been updated according to [Semantic Versioning](http://semver.org/).
- [x] The change has unit & integration tests as appropriate.
- [x] Documentation is up to date and correct.
- [x] Stakeholders have been notified. Workflow-impacting changes have
      been appropriately socialized to avoid surprises.
- [x] The change is either small or have been canaried in the appropriate environment(s).

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -16,31 +16,44 @@
...
     url=REPO_URL,
     download_url=DOWNLOAD_URL,
-    license='License :: OSI Approved :: MIT License',
-    packages=find_packages(),
-    install_requires=REQUIREMENTS,
-    tests_require=[
-        'coverage',
-        'mock',
-        'nose',
+    license='MIT',
```
Comment (dllopis-sfdc) [inline setup.py]: Had we made a decision to continue supporting krux-stdlib as a FOSS project? It seems like a grab-bag of DMP-specific classes & functions.

Comment (hanpeter) [reply setup.py]: As far as I know, it was an MIT licensed project. While whether the other people want this or not may vary, but there is no reason to turn this into a closed source project.

File: setup.py [hunk 2]
```diff
@@ -16,32 +16,43 @@
 # We use the version to construct the DOWNLOAD_URL.
 DOWNLOAD_URL = ''.join((REPO_URL, '/tarball/release/', VERSION))
 
-REQUIREMENTS = ['pystache', 'Sphinx','kruxstatsd', 'argparse', 'lockfile', 'future']
+REQUIREMENTS = ['pystache', 'kruxstatsd', 'lockfile', 'future']
```
Comment (jacobisaliveandwell) [inline setup.py]: Why remove `argparse`?

Comment (hanpeter) [reply setup.py]: Isn't `argparse` already part of Python stdlib? I don't think you need to install it separately.

Comment (jacobisaliveandwell) [reply setup.py]: oh, duh. yeah, that groks.

**Comment by hanpeter** [comment]
@sfdc-slumos & @dllopis-sfdc, rebased the branch to pick up `master`. Please review again.

**Pull request merged**